google/resource_storage_bucket.go

```diff
@@ -282,6 +304,15 @@ func resourceStorageBucketUpdate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if d.HasChange("versioning") {
+		if v, ok := d.GetOk("versioning"); ok {
```

Comment:
Create a new `expandBucketVersioning` method to avoid code duplication and keep consistency with other resources.

See example here: https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_instance_group_manager.go#L740